pkg/claims/run.go

```diff
@@ -70,15 +79,53 @@ func (r Run) DefaultDocumentFilter() interface{} {
 // NewRun creates a run with default values initialized.
 func NewRun(namespace string, installation string) Run {
 	return Run{
-		SchemaVersion: SchemaVersion,
-		ID:            cnab.NewULID(),
-		Revision:      cnab.NewULID(),
-		Created:       time.Now(),
-		Namespace:     namespace,
-		Installation:  installation,
+		SchemaVersion:      SchemaVersion,
+		ID:                 cnab.NewULID(),
+		Revision:           cnab.NewULID(),
+		Created:            time.Now(),
+		Namespace:          namespace,
+		Installation:       installation,
+		ResolvedParameters: make(map[string]interface{}),
+		Parameters:         parameters.NewInternalParameterSet(namespace, installation),
 	}
 }
 
+// PopulateParameters populates the Parameters field with provided parameters.
+// It also saves any sensitive value into the provided secret store.
+func (r *Run) PopulateParameters(params map[string]interface{}, store secrets.Store) error {
+	strategies := make([]secrets.Strategy, 0, len(params))
+	bun := cnab.ExtendedBundle{r.Bundle}
+	for name, value := range params {
+
+		stringVal, err := bun.WriteParameterToString(name, value)
+		if err != nil {
+			return err
+
+		}
+
+		strategy := parameters.DefaultStrategy(name, stringVal)
```

Comment:
I wasn't expecting to need a new Parameter Set resolution strategy for this feature. Can you explain how DefaultStrategy works? It would really help to get a high level walk through of how and where it's used.